### PR TITLE
Feature/nis 3403

### PR DESCRIPTION
**Pull request by Leblyad**

### CODE REVIEW

File: InnoCLinic.OfficesAPI.Core/Exceptions/UserClassExceptions/OfficeNotFoundException.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using MongoDB.Bson;
+
+namespace InnoCLinic.OfficesAPI.Core.Exceptions.UserClassExceptions
+{
+    public class OfficeNotFoundException : NotFoundException
+    {
+        public OfficeNotFoundException(ObjectId objectId) : base($"The office with the identifier {objectId.ToString()} was not found.")
```
Comment (kseniya-kalita) [inline InnoCLinic.OfficesAPI.Core/Exceptions/UserClassExceptions/OfficeNotFoundException.cs:7]: Looks like ToString is redundant

File: InnoClinic.OfficesAPI.Application/DataTransferObjects/OfficeForUpdateDTO.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿namespace InnoClinic.OfficesAPI.Application.DataTransferObjects
+{
+    public class OfficeForUpdateDTO : OfficeForManipulationDTO
```
Comment (kseniya-kalita) [inline InnoClinic.OfficesAPI.Application/DataTransferObjects/OfficeForUpdateDTO.cs:3]: Some fields from US-33 are missing (City, Street, HouseNumber, OfficeNumber). IsActive seems to be unnecessary. Also please add attributes/fluent validation for fields (such as required)

File: InnoClinic.OfficesAPI.Application/DataTransferObjects/OfficeForCreationDTO.cs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+﻿namespace InnoClinic.OfficesAPI.Application.DataTransferObjects
+{
+    public class OfficeForCreationDTO : OfficeForManipulationDTO
```
Comment (kseniya-kalita) [inline InnoClinic.OfficesAPI.Application/DataTransferObjects/OfficeForCreationDTO.cs:3]: Also add missing fields & checks

File: InnoClinic.OfficesAPI.Application/DataTransferObjects/OfficeForCreationDTO.cs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+﻿namespace InnoClinic.OfficesAPI.Application.DataTransferObjects
```
Comment (kseniya-kalita) [inline InnoClinic.OfficesAPI.Application/DataTransferObjects/OfficeForCreationDTO.cs:1]: It's a good idea to create two folders inside DataTransferObjects folder: one for Requests and another one for responses

File: InnoClinic.OfficesAPI.Application/MediatorObjects/Handlers/CreateOfficeHandler.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+using InnoClinic.OfficesAPI.Application.DataTransferObjects;
+using InnoClinic.OfficesAPI.Application.MediatorObjects.Commands;
+using InnoCLinic.OfficesAPI.Core.Contracts.Repositories;
+using InnoCLinic.OfficesAPI.Core.Entities.Models;
+using InnoCLinic.OfficesAPI.Core.Exceptions.UserClassExceptions;
+using MediatR;
+
+namespace InnoClinic.OfficesAPI.Application.MediatorObjects.Handlers
+{
+    public class CreateOfficeHandler : IRequestHandler<OfficeForCreationCommand, OfficeDTO>
```
Comment (kseniya-kalita) [inline InnoClinic.OfficesAPI.Application/MediatorObjects/Handlers/CreateOfficeHandler.cs:11]: Let's store each Handler in the same folder as it's command/query. No need for special place for Handlers

File: InnoClinic.OfficesAPI.Application/MediatorObjects/Commands/OfficeForDeleteCommand.cs [hunk 6]
```diff
@@ -0,0 +1,6 @@
+﻿using MediatR;
+
+namespace InnoClinic.OfficesAPI.Application.MediatorObjects.Commands
+{
+    public record OfficeForDeleteCommand(string officeId) : IRequest;
```
Comment (kseniya-kalita) [inline InnoClinic.OfficesAPI.Application/MediatorObjects/Commands/OfficeForDeleteCommand.cs:5]: Records members should start with an upper case letter

File: InnoClinic.OfficesAPI/InnoClinic.OfficesAPI.csproj [hunk 7]
```diff
@@ -7,12 +7,20 @@
...
+    <PackageReference Include="MediatR.Extensions.Microsoft.DependencyInjection" Version="11.0.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="6.0.11" />
     <PackageReference Include="MongoDB.Driver" Version="2.18.0" />
+    <PackageReference Include="Serilog.AspNetCore" Version="6.1.0" />
     <PackageReference Include="Swashbuckle.AspNetCore" Version="6.2.3" />
   </ItemGroup>
 
   <ItemGroup>
-    <Folder Include="Controllers\" />
+    <ProjectReference Include="..\InnoClinic.OfficesAPI.Application\InnoClinic.OfficesAPI.Application.csproj" />
```
Comment (kseniya-kalita) [inline InnoClinic.OfficesAPI/InnoClinic.OfficesAPI.csproj:21]: Please recheck if we need to reference all this projects. Usually API references only one-two layers. Also check libraries, if they're installed in the dependant projects no need to install this second time here

**Review by kseniya-kalita — COMMENTED** [review]
Good work! Just some small comments to resolve

File: InnoCLinic.OfficesAPI.Core/Exceptions/UserClassExceptions/OfficeNotFoundException.cs (hunk 1 above)
Comment (Leblyad) [reply InnoCLinic.OfficesAPI.Core/Exceptions/UserClassExceptions/OfficeNotFoundException.cs:7]: You right

File: InnoClinic.OfficesAPI.Application/DataTransferObjects/OfficeForUpdateDTO.cs (hunk 2 above)
Comment (Leblyad) [reply InnoClinic.OfficesAPI.Application/DataTransferObjects/OfficeForUpdateDTO.cs:3]: IsActive is Status field, but named by old name. It's required in US's and we need to have possibility to change it in US-32.

File: InnoClinic.OfficesAPI.Application/MediatorObjects/Commands/CreateOfficeHandler.cs [hunk 8]
```diff
@@ -0,0 +1,37 @@
+﻿using AutoMapper;
+using InnoClinic.OfficesAPI.Application.DataTransferObjects;
+using InnoCLinic.OfficesAPI.Core.Contracts.Repositories;
+using InnoCLinic.OfficesAPI.Core.Entities.Models;
+using InnoCLinic.OfficesAPI.Core.Exceptions.UserClassExceptions;
+using MediatR;
+
+namespace InnoClinic.OfficesAPI.Application.MediatorObjects.Commands
+{
+    public class CreateOfficeHandler : IRequestHandler<OfficeForCreationCommand, OfficeDTO>
```
Comment (kseniya-kalita) [inline InnoClinic.OfficesAPI.Application/MediatorObjects/Commands/CreateOfficeHandler.cs:10]: Handler should have the same name as command, e.g. "**CreateOffice**Handler" - "**CreateOffice**Comannd"

File: InnoClinic.OfficesAPI.Application/MediatorObjects/Commands/OfficeForUpdateCommand.cs [hunk 9]
```diff
@@ -0,0 +1,20 @@
+﻿using InnoClinic.OfficesAPI.Application.DataTransferObjects;
+using MediatR;
+
+namespace InnoClinic.OfficesAPI.Application.MediatorObjects.Commands
+{
+    public class OfficeForUpdateCommand : OfficeForManipulation, IRequest
+    {
+        private string Id;
```
Comment (kseniya-kalita) [inline InnoClinic.OfficesAPI.Application/MediatorObjects/Commands/OfficeForUpdateCommand.cs:8]: better just make public string Id {get;set;}

File: InnoClinic.OfficesAPI.Application/MediatorObjects/Commands/OfficeForCreationCommand.cs [hunk 10]
```diff
@@ -0,0 +1,9 @@
+﻿using InnoClinic.OfficesAPI.Application.DataTransferObjects;
+using MediatR;
+
+namespace InnoClinic.OfficesAPI.Application.MediatorObjects.Commands
+{
+    public class OfficeForCreationCommand : OfficeForManipulation, IRequest<OfficeDTO>
```
Comment (kseniya-kalita) [inline InnoClinic.OfficesAPI.Application/MediatorObjects/Commands/OfficeForCreationCommand.cs:6]: better to duplicate code here, not to inherit from OfficeForManipulation

**Review by kseniya-kalita — COMMENTED** [review]
Looks good, just a few comments to resolve and you can merge

**Pull request merged**